orttraining/orttraining/core/graph/optimizer/adam_optimizer_builder.cc

```diff
@@ -52,8 +75,21 @@ Status AdamOptimizerBuilder::Build(
     // In distributed training, some weights may not be updated by all ranks.
     if (opt_configs[i].enabled) {
       // The type proto initializer for Update Count
-      const std::string update_count_string = "Update_Count_" + weight_name;  // per weight optimizer requires a per weight update count
-      TensorProto uc_tensor_proto = CreateTensorProto<int64_t>(update_count_string, 1);
+      std::string uc_prefix = "Update_Count";
+      const std::string update_count_string = uc_prefix + "_" + weight_name;  // per weight optimizer requires a per weight update count
+      TensorProto uc_tensor_proto;
+
+      // Update 'Update_Count' initializer with init value
+      const auto initial_states = opt_configs[i].initial_states;
+      if (initial_states.find(uc_prefix) != initial_states.end()) {
+        const auto& initial_state_it = initial_states.find(uc_prefix);
```

Comment:
may we do one lookup instead of two? Also why is ```initial_state_it``` a reference variable here? Since ```initial_states.find(uc_prefix)``` returns an iterator, I don't see the advantage of using a reference variable here.